Classes_Homework_5/modules/run.py

```diff
@@ -0,0 +1,125 @@
+            elif int(user_num) < int(cnf.get_values("RESTRICTIONS","cal_max")):
+                summary = fitness_tip2
+            else:
+                summary = fitness_tip3
+            return summary
+        except ValueError:
+            raise InvalidNumber(label3, user_num)
+
+    def recipe_values(self):
+        """Pick values for Recipe section"""
+        r_txt = t.ask_for_text(cnf.get_values("INPUTS", "recipe_msg")).strip().capitalize()
+        r_cal = ii.ask_for_int(cnf.get_values("INPUTS", "recipe_calories"))
+        fit_tip = self.get_fitness_message(r_cal)
+        return r_txt, r_cal, fit_tip
+
+    def prepare_for_file(self):
+        """Ask user for selection. Combine values within each section for file + insert to db
+        """
+        selection = s.ask_for_selection(label1, label2, label3)
+        com = Combine(selection, decor, decor_length)
+        if selection == label1:
+            news_city, news_text, news_date = self.news_values()
+            return selection, com.get_news(news_city, news_text, news_date), [news_city, news_text, news_date]
+        elif selection == label2:
+            ad_text, ad_date, formatted = self.ad_values()
+            return selection, com.get_ad(ad_text, ad_date, formatted), [ad_text, formatted]
+        elif selection == label3:
+            rec_txt, rec_cal, fit_tip = self.recipe_values()
+            return selection, com.get_recipe(rec_txt, rec_cal, fit_tip), [rec_txt, rec_cal, fit_tip]
+
+    def write_to_db(self, selection, values):
+        db = DBconnection(cnf.get_values("PATHS", "db_name"))
+        db.create_table(selection)
+        if selection == label1:
+            try:
+                db.go(cnf.get_values("SQL", "insert_news"), (values[0], values[1], values[2]))
+                # lg.write_log(cnf.get_values("MESSAGES", "db_write").format
+                #              (s=selection, t=selection.lower(), v1=values[0], v2=values[1], src="manual input"))
+            except sql.IntegrityError:
+                raise Duplicate(selection, values[0], values[1])
+        elif selection == label2:
+            try:
+                db.go(cnf.get_values("SQL", "insert_ad"), (values[0], values[1]))
+                # lg.write_log(cnf.get_values("MESSAGES", "db_write").format
+                #              (s=selection, t=selection.lower(), v1=values[0], v2=values[1], src="manual input"))
+            except sql.IntegrityError:
+                raise Duplicate(selection, values[0], values[1])
+        elif selection == label3:
+            try:
+                db.go(cnf.get_values("SQL", "insert_recipe"), (values[0], values[1], values[2]))
+                # lg.write_log(cnf.get_values("MESSAGES", "db_write").format
+                #              (s=selection, t=selection.lower(), v1=values[0], v2=values[1], src="manual input"))
+            except sql.IntegrityError:
+                raise Duplicate(selection, values[0], values[1])
```

Comment:
Your code is basically the same for each type of selection. How about the next more unified example?
```
        bindings = ','.join(list('?' * len(values)))
        try:
            db.go(f'INSERT INTO {selection} VALUES ({bindings})', values)
        except sql.IntegrityError:
            raise Duplicate(selection, *values)
```
Then these lines can be deleted in the config.ini file:
```
insert_news = INSERT INTO news VALUES(?,?,?)
insert_ad = INSERT INTO ad VALUES(?,?)
insert_recipe = INSERT INTO recipe VALUES(?,?,?)
```

Reply:
Also, I think it'd be better to create a separate function for inserting. For example, it'd be useful if we want only to create tables, not make inserts.